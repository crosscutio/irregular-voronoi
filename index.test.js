"use strict";

const irregularVoronoi = require("./index");
const { writeFileSync } = require("fs");

test("works", () => {
  const input = require("./__fixtures__/input.json");

  const polygon = input[0];
  const points = input.slice(1);

  const polys = irregularVoronoi(polygon, points);
  const fc = {
    type: "FeatureCollection",
    features: polys.concat(points),
  };

  writeFileSync("./__fixtures__/result.json", JSON.stringify(fc));

  expect(polys).toMatchSnapshot();
});
