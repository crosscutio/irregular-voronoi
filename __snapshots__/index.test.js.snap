// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
Array [
  Object {
    "geometry": Object {
      "coordinates": Array [
        Array [
          Array [
            31.10401153564453,
            -26.313728134785585,
          ],
          Array [
            31.115042609032113,
            -26.312550975886914,
          ],
          Array [
            31.11359462229664,
            -26.28801794348307,
          ],
          Array [
            31.10401153564453,
            -26.288336165488044,
          ],
          Array [
            31.10469818115234,
            -26.299570874272394,
          ],
          Array [
            31.112079620361325,
            -26.29757025236952,
          ],
          Array [
            31.111907958984375,
            -26.306188069982834,
          ],
          Array [
            31.10418319702148,
            -26.307111369583296,
          ],
          Array [
            31.10401153564453,
            -26.313728134785585,
          ],
        ],
      ],
      "type": "Polygon",
    },
    "properties": Object {},
    "type": "Feature",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        Array [
          Array [
            31.118160681067902,
            -26.28786632006258,
          ],
          Array [
            31.127859028653045,
            -26.296372098021553,
          ],
          Array [
            31.12797674228375,
            -26.30157220310172,
          ],
          Array [
            31.130962371826172,
            -26.311881633667724,
          ],
          Array [
            31.137828826904297,
            -26.310958372073717,
          ],
          Array [
            31.139992398255462,
            -26.2988894681617,
          ],
          Array [
            31.135797680498715,
            -26.298019169343384,
          ],
          Array [
            31.134910583496094,
            -26.304033675643918,
          ],
          Array [
            31.127872467041012,
            -26.286489259881638,
          ],
          Array [
            31.12255096435547,
            -26.2877205335534,
          ],
          Array [
            31.118160681067902,
            -26.28786632006258,
          ],
        ],
      ],
      "type": "Polygon",
    },
    "properties": Object {},
    "type": "Feature",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        Array [
          Array [
            31.11359462229664,
            -26.28801794348307,
          ],
          Array [
            31.115042609032113,
            -26.312550975886914,
          ],
          Array [
            31.125640869140625,
            -26.31142000379029,
          ],
          Array [
            31.12392425537109,
            -26.30464922096988,
          ],
          Array [
            31.116371154785153,
            -26.30541864803031,
          ],
          Array [
            31.115341186523438,
            -26.296800773214482,
          ],
          Array [
            31.126327514648438,
            -26.295877391487554,
          ],
          Array [
            31.12797674228375,
            -26.30157220310172,
          ],
          Array [
            31.127859028653045,
            -26.296372098021553,
          ],
          Array [
            31.118160681067906,
            -26.28786632006258,
          ],
          Array [
            31.11359462229664,
            -26.28801794348307,
          ],
        ],
      ],
      "type": "Polygon",
    },
    "properties": Object {},
    "type": "Feature",
  },
  Object {
    "geometry": Object {
      "coordinates": Array [
        Array [
          Array [
            31.135797680498715,
            -26.298019169343384,
          ],
          Array [
            31.139992398255462,
            -26.2988894681617,
          ],
          Array [
            31.14246368408203,
            -26.285104061378636,
          ],
          Array [
            31.137657165527344,
            -26.28541188469766,
          ],
          Array [
            31.135797680498715,
            -26.298019169343384,
          ],
        ],
      ],
      "type": "Polygon",
    },
    "properties": Object {},
    "type": "Feature",
  },
]
`;
